tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -14,33 +14,90 @@ else
 }
 
 
-function Get-AzProfile 
+function Get-ProfileCachePath
 {
-  @{
-    '2015-05' = 
-    @{
-        'AzureRM' = @('1.2.4');
-        'AzureRM.Storage' = @('1.2.4');
-        'Azure.Storage' = @('1.2.4');
-        'AzureRM.Profile' = @('1.2.4');
-     }; 
-     '2016-09' = 
-     @{
-        'AzureRM' = @('2.0.1');
-        'AzureRM.Storage' = @('2.0.1')
-        'Azure.Storage' = @('2.0.1');
-        'AzureRM.Profile' = @('2.0.1');
-     }
-   }
+  $ProfileCache = Join-Path -path $env:LOCALAPPDATA -childpath "Microsoft\AzurePowerShell\ProfileCache"
+  return $ProfileCache
+}
+
+
+# Get-ProfileMap from Azure Endpoint
+function Get-AzureProfileMap
+{
+  $ProfileCache = Get-ProfileCachePath
```

Comment:
We already have a default profile map that ships with the cmdlets. Please check lines 83 & 84. 
The caching works like this: 
(I) Get-AzProfile first looks for ProfileMap in cache
(ii) If its not available, then looks at the default ProfileMap ($PSCommandPath)
(iii) If Get-AzProfile is used with -Force, then updates the cache from online. It doesn't prompt for update unless specified with -Force. (I'll change -Force to -Online)